### ISSUE TITLE
Command Input missing uppercase letters

### ISSUE THREAD

**Issue by schrc3b6** [vim/vim#5612]
Hi,

I noticed that my vim ignores uppercase letters in terminal commands.
For example if I run `:w !sudo tee % > /dev/null` it only works for lowercase passwords.
Also Plugins like fzf.vim are affected.

I also ran the following script in vim to see which inputs are registered:
```
#!/usr/bin/env python
from getpass import getpass

pw = getpass("pw: ")
print("pw '%s'" % (pw))
```
Only lowercase characters were printed. (I also tried it with --clean)

I also noticed that if I hit a uppercase character while a command is running (for example the sudo write), I see the following string for an A: `^[[27;2;65~`

I'm using vim from the arch repository: vim-runtime-8.2.0148-1
```
:version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jan 26 2020 11:45:38)
Included patches: 1-148
Compiled by Arch Linux
Huge version with GTK3 GUI.  Features included (+) or not (-):
+acl               +channel           +cscope            +ex_extra          -hangul_input      +lispindent        +mouse_gpm         +netbeans_intg     +python/dyn        +spell             +terminal          +vertsplit         +X11
+arabic            +cindent           +cursorbind        +extra_search      +iconv             +listcmds          -mouse_jsbterm     +num64             +python3/dyn       +startuptime       +terminfo          +virtualedit       -xfontset
+autocmd           +clientserver      +cursorshape       -farsi             +insert_expand     +localmap          +mouse_netterm     +packages          +quickfix          +statusline        +termresponse      +visual            +xim
+autochdir         +clipboard         +dialog_con_gui    +file_in_path      +job               +lua/dyn           +mouse_sgr         +path_extra        +reltime           -sun_workshop      +textobjects       +visualextra       -xpm
-autoservername    +cmdline_compl     +diff              +find_in_path      +jumplist          +menu              -mouse_sysmouse    +perl/dyn          +rightleft         +syntax            +textprop          +viminfo           +xsmp_interact
+balloon_eval      +cmdline_hist      +digraphs          +float             +keymap            +mksession         +mouse_urxvt       +persistent_undo   +ruby/dyn          +tag_binary        +timers            +vreplace          +xterm_clipboard
+balloon_eval_term +cmdline_info      +dnd               +folding           +lambda            +modify_fname      +mouse_xterm       +popupwin          +scrollbind        -tag_old_static    +title             +wildignore        -xterm_save
+browse            +comments          -ebcdic            -footer            +langmap           +mouse             +multi_byte        +postscript        +signs             -tag_any_white     +toolbar           +wildmenu
++builtin_terms    +conceal           +emacs_tags        +fork()            +libcall           +mouseshape        +multi_lang        +printer           +smartindent       +tcl/dyn           +user_commands     +windows
+byte_offset       +cryptv            +eval              +gettext           +linebreak         +mouse_dec         -mzscheme          +profile           +sound             +termguicolors     +vartabs           +writebackup
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "/etc/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/lib/libffi-3.2.1/include -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/inclu
de/freetype2 -I/usr/include/libpng16 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/a
t-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -pthread    -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_REENTRANT  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -L. -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector-strong -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.30/core_perl/CORE  -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -L/usr/local/lib -Wl,--as-n
eeded -o vim   -lgtk-3 -lgdk-3 -lz -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0  -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lelf -lnsl    -lcanberra  -lacl -lattr
 -lgpm -ldl   -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.30/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/perl5/5.30/core_perl/CORE -lperl -lpthread -ldl -lm -lcrypt -lutil -lc   -L/usr/lib
-ltclstub8.6 -ldl -lz -lpthread -lm
```

Thanks for any help in advance!
Max

**Comment by k-takata** [comment]
Looks like modifyOtherKeys caused the issue. (See `:help modifyOtherKeys`)
Which terminal emulator do you use?

**Comment by schrc3b6** [comment]
I'm using xterm (353).
If I set `XTerm*vt100.modifyOtherKeys: 1` in my Xresources and 
disable modifyOtherKeys in my vimrc everything works again.

But shouldn't it also work the other way around or did I misunderstood the help pages?

**Comment by antofthy** [comment]
A temporary fix... see https://github.com/vim/vim/issues/5200

**Comment by brammool** [comment]
> A temporary fix... see https://github.com/vim/vim/issues/5200

Vim relies on being able to change unmapped sequences to the regular key
somewhere.  For what you are doing that code isn't used.  The Python
getpass() function is a good example.

I think the only way is to let the plugin temporarily disable
modifyOtherKeys.  Using a function is the most flexible:

	call modify_other_keys(v:false)  " disable modifyOtherKeys
	call modify_other_keys(v:true)   " enable modifyOtherKeys

In case a plugin breaks, we can tell the user to execute a shell command
to get back to normal: !dir

-- 
hundred-and-one symptoms of being an internet addict:
55. You ask your doctor to implant a gig in your brain.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by antofthy** [comment]
@brammool I get an error "E117: Unknown function: modify_other_keys"

**Comment by k-takata** [comment]
Obviously, it is not implemented yet.

**Comment by chrisbra** [comment]
fixed by 4132eb505c506f7a0710b03a275d1c5a247e4196

**Comment by antofthy** [comment]
FYI.  Patch is to allow use of `echoraw(&te_TI)` to disable ModifyOtherKeys on the tty.
See  https://github.com/vim/vim/issues/5617  for more info.
